lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -534,7 +534,6 @@ def add_metadata(post, options = {})
         post[:metadata].merge!(options[:metadata]) if options[:metadata]
         post[:metadata][:email] = options[:email] if options[:email]
         post[:metadata][:order_id] = options[:order_id] if options[:order_id]
-        post.delete(:metadata) if post[:metadata].empty?
```

Comment:
This line was removed because:
* It is superfluous. There appears to be no problem adding an empty `metadata` object to the request.
* The Stripe PI gateway uses this method, and this line was precluding us from extending the method easily.